jobrunner/service.py

```diff
@@ -32,7 +32,7 @@ def main():
     configure_logging(fmt)
     # ensure db is set up before we start any threads, otherwise if we don't already have
     # a database (which we don't in tests) then threads will race to create the schema
-    get_connection()
+    init_db()
```

Comment:
This call means that if the database file isn't in the place we expect it we'll just create a new empty file and carry on without error. But I thought the idea was that we'd fail hard unless the database file had already been created?

Reply:
Hmm, yes we did discuss that. But my main concern was to detect when migrations needed applying, so that got dropped. And ever

It's a bit messy, as it means init_db will need two modes when a db doesn't exist one where it fails (which is only used here) and one where it doesn't (which is used everywhere else, i.e. tests and local_run.

And how do we know if this is a new backend or not? We'd need a way to create the initial db so this check wouldn't error.  The current migrate command doesn't do that, it assumes a db exists. 

So, if we use django as our model, then an initial migrate *is* needed before a service will run successfully in a new environment. So perhaps the migrate command should do that.

We'll still need to two modes though, or possibly two different functions.

Reply:
Have updated to implement the above